### ISSUE TITLE
Json4s brings in scalap-2.10.0 dependencies regardless of scalaVersion setting

### ISSUE THREAD

**Issue by mcamou** [json4s/json4s#108]
With the following build.sbt:

```
val json4sVersion = "3.2.7"

name := "Json4sDependencies"

organization := "com.tecnoguru"

version := "1.0"

scalaVersion := "2.10.3"

libraryDependencies ++= Seq("org.json4s" %% "json4s-jackson" % json4sVersion,
                            "org.json4s" %% "json4s-ext" % json4sVersion)
```

I get the following inside SBT:

```
> show dependencyClasspath
[info] List(Attributed(/Users/mario/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.10.3.jar), Attributed(/Users/mario/.ivy2/cache/org.json4s/json4s-jackson_2.10/jars/json4s-jackson_2.10-3.2.7.jar), Attributed(/Users/mario/.ivy2/cache/org.json4s/json4s-core_2.10/jars/json4s-core_2.10-3.2.7.jar), Attributed(/Users/mario/.ivy2/cache/org.json4s/json4s-ast_2.10/jars/json4s-ast_2.10-3.2.7.jar), Attributed(/Users/mario/.ivy2/cache/com.thoughtworks.paranamer/paranamer/jars/paranamer-2.6.jar), Attributed(/Users/mario/.ivy2/cache/org.scala-lang/scalap/jars/scalap-2.10.0.jar), Attributed(/Users/mario/.ivy2/cache/org.scala-lang/scala-compiler/jars/scala-compiler-2.10.0.jar), Attributed(/Users/mario/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.10.0.jar), Attributed(/Users/mario/.ivy2/cache/com.fasterxml.jackson.core/jackson-databind/bundles/jackson-databind-2.3.1.jar), Attributed(/Users/mario/.ivy2/cache/com.fasterxml.jackson.core/jackson-annotations/bundles/jackson-annotations-2.3.0.jar), Attributed(/Users/mario/.ivy2/cache/com.fasterxml.jackson.core/jackson-core/bundles/jackson-core-2.3.1.jar), Attributed(/Users/mario/.ivy2/cache/org.json4s/json4s-ext_2.10/jars/json4s-ext_2.10-3.2.7.jar), Attributed(/Users/mario/.ivy2/cache/joda-time/joda-time/jars/joda-time-2.3.jar), Attributed(/Users/mario/.ivy2/cache/org.joda/joda-convert/jars/joda-convert-1.6.jar))
[success] Total time: 0 s, completed Mar 20, 2014 3:18:47 PM
```

Note the scalap-2.10.0, scala-compiler-2.10.0 and scala-reflect-2.10.0 dependencies. Using the sbt-dependency-graph plugin I've narrowed it down to the scalap dependency. You can work around this by adding an explicit dependency on `"org.scala-lang" % "scalap" % scalaVersion.value` in your build.sbt but this should really be fixed in Json4s

**Comment by jongwook** [comment]
Apache Spark uses json4s and this issue broke my scala 2.11 project build.

I had to explicitly exclude scalap and scala-library like

```scala
libraryDependencies ++= Seq(
   // dependencies ... 
).map {
    _.exclude("org.scala-lang", "scalap").exclude("org.scala-lang", "scala-library")
}
```

otherwise sbt gets confused by the scala-library pulled by json4s, and IntelliJ won't even let me open the project.

Is it fixed in 3.3.0?

**Comment by seratch** [comment]
>Is it fixed in 3.3.0?

Yes, it is. This issue doesn't happen in 3.3 because json4s 3.3 has its own scalap fork and doesn't depend on scala-lang's scalap.
